contrib/pkg/v1migration/deleteobjects.go

```diff
@@ -90,68 +93,54 @@ func (o *DeleteObjectsOptions) Run() error {
 			}
 			return errors.Wrap(err, "could not decode JSON from file")
 		}
-		o.deleteObject(client, &objFromFile)
+		objChan <- &objFromFile
 	}
+	close(objChan)
+	deleteWG.Wait()
 	return nil
 }
 
-func (o *DeleteObjectsOptions) deleteObject(client dynamic.Interface, objFromFile *unstructured.Unstructured) {
-	apiVersion := objFromFile.GetAPIVersion()
-	kind := objFromFile.GetKind()
-	namespace := objFromFile.GetNamespace()
-	name := objFromFile.GetName()
-	logger := log.WithFields(log.Fields{
-		"apiVersion": apiVersion,
-		"kind":       kind,
-		"name":       name,
-	})
-	if namespace != "" {
-		logger = logger.WithField("namespace", namespace)
-	}
-	if apiVersion != hivev1alpha1.SchemeGroupVersion.String() {
-		logger.Warn("object in JSON file is not a Hive v1alpha1 resource")
-		return
-	}
-	gvr := hivev1alpha1.SchemeGroupVersion.WithResource(resourceForHiveKind(kind))
-	var resourceClient dynamic.ResourceInterface
-	if namespace != "" {
-		resourceClient = client.Resource(gvr).Namespace(namespace)
-	} else {
-		resourceClient = client.Resource(gvr)
-	}
-	objFromKube, err := resourceClient.Get(name, metav1.GetOptions{})
-	switch {
-	case apierrors.IsNotFound(err):
-		logger.Warn("object not found")
-		return
-	case err != nil:
-		logger.WithError(err).Error("failed to get object")
-		return
-	}
-
-	// Ignore owner refs and resource version when looking for diffs as some Hive objects may have been orphaned.
-	// TODO: Check changes to owner refs to ensure that the only changes are removals of references to Hive objects.
-	objFromFile.SetOwnerReferences(objFromKube.GetOwnerReferences())
-	objFromFile.SetResourceVersion(objFromKube.GetResourceVersion())
-
-	// For cluster-scoped resources, set the namespace to the empty string so that the deep equal check does not pick
-	// up a difference in the namespace. Setting to an empty string will delete the .metadata.namespace entry from the
-	// unstructured data.
-	if namespace == "" {
-		objFromFile.SetNamespace("")
-	}
+func deleteObjects(client dynamic.Interface, objChan chan *unstructured.Unstructured, wg *sync.WaitGroup, logger log.FieldLogger) {
+	for objFromFile := range objChan {
+		apiVersion := objFromFile.GetAPIVersion()
+		kind := objFromFile.GetKind()
+		namespace := objFromFile.GetNamespace()
+		name := objFromFile.GetName()
+		logger := logger.WithFields(log.Fields{
```

Comment:
Since I took the time to look it up, I thought I'd share here that logrus "is protected by a mutex for concurrent writes."

From: https://github.com/sirupsen/logrus/blob/master/README.md#thread-safety